.github/workflows/analysis.yml

```diff
@@ -0,0 +1,40 @@
+name: Static Analysis
+on: [push, pull_request]
+permissions:
+  contents: read
+
+jobs:
+  static_analysis:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - uses: actions/setup-go@v5
+      with:
+        go-version: 'stable'
+
+    - name: Install analysis tools
+      run: |
+        go install golang.org/x/tools/cmd/goimports@latest
+        go install github.com/fzipp/gocyclo/cmd/gocyclo@latest
+        go install honnef.co/go/tools/cmd/staticcheck@latest
+
+    - name: Prepare
+      run: rm -rf v3.0/ v3.1/ v3.2/
+
+    - name: Get dependencies
+      run: sudo apt-get update && sudo apt-get install gcc libgl1-mesa-dev libegl1-mesa-dev libgles2-mesa-dev libx11-dev xorg-dev
+
+    - name: Vet
+      run: go vet ./...
+
+    - name: Goimports
+      run: test -z "$(goimports -e -d . | tee /dev/stderr)"
```

Comment:
I'm not sure how helpful it is to use goimports instead of using gofmt. I think gofmt would be helpful and sufficient.

But if you think there's enough upsides to goimports, let's keep it as is for now. Up to you.